File: src/zk/hashing.js
/**
 * NeuroToken ZK System — Hashing Module
 * Provides a  Poseidon-like hash using SHA-256
 * (safe for open-source, no sensitive ZK construction)
 */
import crypto from "crypto";

export function nt_hash(inputs) {
  if (!Array.isArray(inputs)) throw new Error("inputs must be array");
  const data = inputs.map(x => x.toString()).join("|");
  return crypto.createHash("sha256").update(data).digest("hex");
}

File: src/zk/hashing.py
# NeuroToken ZK System — Hashing Module (Python)
# Public-safe  using SHA-256.

import hashlib

def nt_hash(inputs):
    if not isinstance(inputs, list):
        raise ValueError("inputs must be list")
    data = "|".join([str(x) for x in inputs])
    return hashlib.sha256(data.encode()).hexdigest()
